packages/trace-viewer/src/snapshotRenderer.ts

```diff
@@ -66,6 +66,9 @@ export class SnapshotRenderer {
         } else if (typeof n[0] === 'string') {
           // Element node.
           const builder: string[] = [];
+          // mask noscript tags in trace-viewers HTML display
+          if (n[0] === 'NOSCRIPT')
```

Comment:
Instead of post-processing, I think we should just [write a different `nodeName`](https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts#L388) into tracing right away, because that's the only place where we know whether `noscript` should be rendered or not.

```ts
const result: NodeSnapshot = [nodeName === 'NOSCRIPT' ? 'MASKED-NOSCRIPT' : nodeName, attrs];
```

Reply:
Hmmm -- I did it in the trace viewer, because it's a flaw caused by the used display type. if we wouldn't use HTML-widgets for the output of the report, everything would be fine and perfectly undistorted without this tricky workaround. So it's IMHO indeed mainly a viewer issue and should be handled there.